src/fieldValidation.ts

```diff
@@ -11,11 +26,9 @@ export const fieldValueIsValid = (
 		case BaseFieldDataType.PHONE_NUMBER:
 			return phone(fieldValue).isValid;
 		case BaseFieldDataType.EMAIL:
-			return /^\S+@\S+$/.test(fieldValue);
+			return ajv.validate(emailSchema, fieldValue);
```

Comment:
I'm torn on having the email / uri schemas removed from the context where the validation itself happens...

On the other hand, well named functions / variables are really helpful...

WHAT DO YOU THINK OF........

Defining a bunch of helper validation functions and calling those directly:

`isValidEmail`, `isValidPhoneNumber`, `isValidUrl` -- and then having all the actual implementations in there.  That way the `fieldValueIsValid` has absolutely no implementation whatsoever, and instead just routes the validation call to the appropriately scoped function -- this way the only time we edit this function is when adding a new format, but if the implementation of any given validator changes, we edit that validator.

Eh? eh?

Reply:
sure! seems like a more scalable approach. 
Two questions:
1. Should those functions be defined in the same file(`fieldValidation.ts`)
2. Should the schemas be defined in the functions or outside the functions

Reply:
Ok did a refactor, let me know whatcha think, will be good to figure this out now so i can follow the pattern in issue #899

Reply:
Good question!  Either is a fine approach at this point and might come down to personal taste; it feels a little verbose to have them in separate files BUT that is the pattern we've been using.  I tend to think "many files with well defined scope is better than fewer files with unbound scopes" is the way to go.

That said, having the definitions in the immediate context they're used is totally reasonable, and in this case these functions are ONLY used in that single context.

I'd probably put them in one file for now (and not exported), and if some day they get re-used OR the file gets hard to work with we can always break it into smaller pieces; that said if you'd prefer to break it up right away in the name of long term scalability / avoiding a future of a mega-file then I fully support that!